### PR TITLE
Add new rewrite rule:  CamelCaseVariableNaming

### PR DESCRIPTION
**Pull request by mggger**
Different developers have different variable naming styles, the most common is:

1.  SnakeCase  ```val student_name = "xxx" ```
2. CamelCase ```val studentName = "xxx" ```

Add new rewrite rule **CamelCaseVariableNaming** to unify variable naming style, here is a example :

```scala
class A {
  val student_name = "wang"
  val studentAge = 22
}

// after formatting
// rewrite.rules = [CamelCaseVariableNaming]
class A {
  val studentName = "wang"
  val studentAge = 22
}

```

### CODE REVIEW

File: scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/CamelCaseVariableNaming.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import scala.meta.{Pat, Tree}
+
+object CamelCaseVariableNaming extends Rewrite {
+  override def create(implicit ctx: RewriteCtx): RewriteSession =
+    new CamelCaseVariableNaming
+}
+
+class CamelCaseVariableNaming(implicit ctx: RewriteCtx) extends RewriteSession {
+
+  def toCamelCaseNaming(str: String): String = {
```
Comment (poslegm) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/CamelCaseVariableNaming.scala:12]: Variable names in Scala may start with underscore:
```scala
val _abc = 1
```

And this function works wrong on it:
```
@ toCamelCaseNaming("_abc")
res2: String = "Abc"
```

File: scalafmt-tests/src/test/resources/rewrite/CamelCaseVariableNaming.stat [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  val studentSex = "male"
+}
+>>>
+class A {
+  val studentName = "wang"
+  val studentAge = 12
+  val studentSex = "male"
+}
+
+
```
Comment (poslegm) [inline scalafmt-tests/src/test/resources/rewrite/CamelCaseVariableNaming.stat:16]: Would rewrite rule work correct on provided tests?

```scala
<<< rewrite also identifier usages
val x = 42 + snake_case_val
>>>
val x = 42 + snakeCaseVal
<<< don't rewrite names from external libraries
another.library.ExternalClass.snake_case_method_name(42)
>>>
another.library.ExternalClass.snake_case_method_name(42)
```

**Comment by mggger** [comment]
@poslegm Thanks  for your code reviewing, it would be very helpful for me .

@kitbellew Thanks,  there are a lot of flaws in my implementation

**Comment by kitbellew** [comment]
@mggger perhaps you could use `scalastyle` to check compliance (for instance, http://www.scalastyle.org/rules-1.0.0.html#org_scalastyle_scalariform_FieldNamesChecker)

**Pull request closed without merging**